plugins/pulp_rpm/plugins/importers/yum/listener.py

```diff
@@ -196,6 +196,7 @@ def download_succeeded(self, report):
                 return
             added_unit = self.sync.add_rpm_unit(self.metadata_files, unit)
```

Comment:
@jeremycline, these are really great points!
Issue 1: Right now between this call and `self.sync.associate_rpm_unit` there is only a `safe_import_content` call which [cleans orphan in case of any exception](https://github.com/pulp/pulp/blob/2.9-release/server/pulp/server/db/model/__init__.py#L907). But It does not prevent someone to add more code in between and have the orphan at the end :( I am not sure if it is good to add another try-except here and ask to put code into this statement in future.
As for the empty file, my next PR is trying to prevent empty or corrupted files to appear at the `unit.storage_path` (at least for RPMs).

Issue 2: I agree that general logic of when-to-sync can be improved but this PR does not change this behavior anyhow, so maybe it is  better to make this improvement outside of this PR.